analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -88,19 +136,66 @@ public Verifier(VerifierBuilder builder)
             }
         }
 
-        private IEnumerable<Compilation> Compile()
+        public void VerifyCodeFix()     // This should never has any arguments
+        {
+            _ = codeFix ?? throw new InvalidOperationException($"{nameof(builder.CodeFix)} was not set.");
+            var document = CreateProject(false).FindDocument(Path.GetFileName(builder.Paths.Single()));
+            var codeFixVerifier = new CodeFixVerifier(analyzers.Single(), codeFix, document, builder.CodeFixTitle);
+            var fixAllProvider = codeFix.GetFixAllProvider();
+            foreach (var parseOptions in builder.ParseOptions.OrDefault(language.LanguageName))
+            {
+                codeFixVerifier.VerifyWhileDocumentChanges(parseOptions, builder.CodeFixedPath);
+                if (fixAllProvider is not null)
+                {
+                    codeFixVerifier.VerifyFixAllProvider(fixAllProvider, parseOptions, builder.CodeFixedPathBatch ?? builder.CodeFixedPath);
+                }
+            }
+        }
+
+        public void VerifyUtilityAnalyzerProducesEmptyProtobuf()     // This should never has any arguments
+        {
+            foreach (var compilation in Compile(false))
+            {
+                DiagnosticVerifier.Verify(compilation, analyzers.Single(), CompilationErrorBehavior.Default);
+                new FileInfo(builder.ProtobufPath).Length.Should().Be(0, "protobuf file should be empty");
+            }
+        }
+
+        public void VerifyUtilityAnalyzer<TMessage>(Action<IReadOnlyList<TMessage>> verifyProtobuf)
```

Comment:
This breaks the `no arguments` pattern :/ but other ways are more complicated and would move this verifier functionality out of this class.